bindings/java/iota-client-java/src/main/java/org/iota/apis/BaseApi.java

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+package org.iota.apis;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.commons.lang3.SystemUtils;
+import org.iota.types.ClientConfig;
+import org.iota.types.ClientException;
+
+import java.io.IOException;
+
+public class BaseApi {
+
+    protected ClientConfig clientConfig;
+
+    protected BaseApi(ClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+    }
+
+    static {
```

Comment:
Should this really be done in the base API ?

Reply:
Wait actually what is the Base API? The name confused me, is that the message passing ?